_posts/2020-02-03-this-week-in-glean.md

```diff
@@ -0,0 +1,145 @@
+
+The last blog post: [This Week in Glean: Glossary](https://brizental.github.io/2020/01/10/this-week-in-glean-glossary.html) by Bea.
+
+---
+
+As :chutten outlined in the first [TWiG blog post](https://chuttenblog.wordpress.com/2019/10/17/this-week-in-glean-glean-on-desktop-project-fog/) we're currently prototyping Glean on Desktop.
+After a couple rounds of review, some adjustements and some learnings from doing Rust on mozilla-central, we were ready to land the first working prototype code earlier this year ([Bug 1591564][bz-landing]).
+
+Unfortunately the patch set was [backed out nearly immediately][mc-backout] [^1] for 2 failures.
+The first one was a "leak" (we missed cleaning up memory in a way to satisfy the rigorous Firefox test suite, that was fixed in [another patch](https://phabricator.services.mozilla.com/D59531)).
+The second one was a build failure on a Windows platform.
+
+This is what the log had to say about it (shortened to the relevant parts here, [see the full log output][logoutput]):
+
+```
+lld-link: error: undefined symbol: __rbt_backtrace_pcinfo
+>>> referenced by gkrust_gtest.lib(backtrace-5286ea09b9822175.backtrace.3kzojw1m-cgu.3.rcgu.o)
+lld-link: error: undefined symbol: __rbt_backtrace_create_state
+>>> referenced by gkrust_gtest.lib(backtrace-5286ea09b9822175.backtrace.3kzojw1m-cgu.3.rcgu.o)
+lld-link: error: undefined symbol: __rbt_backtrace_syminfo
+>>> referenced by gkrust_gtest.lib(backtrace-5286ea09b9822175.backtrace.3kzojw1m-cgu.3.rcgu.o)
+clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
+/builds/worker/workspace/build/src/config/rules.mk:608: recipe for target 'xul.dll' failed]
+```
+
+I set out to investigate this error.
+While I had not seen that particular error before, I knew about the [`backtrace`][backtrace] crate. It caused me some trouble before (it depends on a C library, and won't work on all targets easily).
+I knew that Glean doesn't really depend on its functionality[^2] and thus removing it from our dependency graph would probably solve the issue.
+But first I had to find out why we depend on it somewhere and why it is causing these linker errors to begin with.
+
+The first thing I noticed is that we didn't include anything new in the patch set that was now rejected.
+Through some experimentation and use [`cargo-tree`][] I could tell that `backtrace` was included in the build before our Glean patch[^3], as a transitive dependency of another crate: [`failure`][].
+
+So why didn't it fail the build before?
+As per the errors above the build failed only during linking, not compilation, which makes me believe those functions where never linked in previously, because no one passed around any errors that would cause these functions to be used.
```

Comment:
```suggestion
As per the errors above, the build failed only during linking, not compilation, which makes me believe those functions where never linked in previously, because no one passed around any errors that would cause these functions to be used.
```

Reply:
Oh geez, and I missed
```patch
-those functions where never linked in
+those functions were never linked in
```